src/routes/accounts/counterfactual-safes/counterfactual-safes.controller.spec.ts

```diff
@@ -400,6 +453,316 @@ describe('CounterfactualSafesController', () => {
+      expect(accountsRepository.getAccount).not.toHaveBeenCalled();
+      expect(
+        counterfactualSafesDataSource.getCounterfactualSafesForAccount,
+      ).not.toHaveBeenCalled();
+    });
+
+    it('returns 403 is token it not yet valid', async () => {
+      const address = getAddress(faker.finance.ethereumAddress());
+      const chain = chainBuilder().build();
+      const authPayloadDto = authPayloadDtoBuilder()
+        .with('chain_id', chain.chainId)
+        .with('signer_address', address)
+        .build();
+      const accessToken = jwtService.sign({
+        ...authPayloadDto,
+        nbf: faker.date.future(),
+      });
+
+      await request(app.getHttpServer())
+        .get(`/v1/accounts/${address}/counterfactual-safes`)
+        .set('Cookie', [`access_token=${accessToken}`])
+        .expect(403);
+
+      expect(() => jwtService.verify(accessToken)).toThrow('jwt not active');
+      expect(accountsRepository.getAccount).not.toHaveBeenCalled();
+      expect(
+        counterfactualSafesDataSource.getCounterfactualSafesForAccount,
+      ).not.toHaveBeenCalled();
+    });
+
+    it('returns 403 if token has expired', async () => {
+      const address = getAddress(faker.finance.ethereumAddress());
+      const chain = chainBuilder().build();
+      const authPayloadDto = authPayloadDtoBuilder()
+        .with('chain_id', chain.chainId)
+        .with('signer_address', address)
+        .build();
+      const accessToken = jwtService.sign({
+        ...authPayloadDto,
+        exp: faker.date.past(),
+      });
+
+      expect(() => jwtService.verify(accessToken)).toThrow('jwt expired');
+      await request(app.getHttpServer())
+        .get(`/v1/accounts/${address}/counterfactual-safes`)
+        .set('Cookie', [`access_token=${accessToken}`])
+        .expect(403);
+
+      expect(accountsRepository.getAccount).not.toHaveBeenCalled();
+      expect(
+        counterfactualSafesDataSource.getCounterfactualSafesForAccount,
+      ).not.toHaveBeenCalled();
+    });
+
+    it('returns 403 if signer_address is not a valid Ethereum address', async () => {
+      const address = getAddress(faker.finance.ethereumAddress());
+      const chain = chainBuilder().build();
+      const authPayloadDto = authPayloadDtoBuilder()
+        .with('chain_id', chain.chainId)
+        .with('signer_address', faker.string.sample() as `0x${string}`)
+        .build();
+      const accessToken = jwtService.sign(authPayloadDto);
+
+      await request(app.getHttpServer())
+        .get(`/v1/accounts/${address}/counterfactual-safes`)
+        .set('Cookie', [`access_token=${accessToken}`])
+        .expect(403);
+
+      expect(accountsRepository.getAccount).not.toHaveBeenCalled();
+      expect(
+        counterfactualSafesDataSource.getCounterfactualSafesForAccount,
+      ).not.toHaveBeenCalled();
+    });
+
+    it('returns 403 if chain_id is not a valid chain ID', async () => {
```

Comment:
What do you think about adding a tests to check the validity of a token from a different chain for these endpoints?

Reply:
I think that should be covered by the base case ([link](https://github.com/safe-global/safe-client-gateway/blob/d503281efc6a0d780e810b1c4093668632e6ae0d/src/routes/accounts/counterfactual-safes/counterfactual-safes.controller.spec.ts#L459)) as the `AuthPayload` chain ID is not connected in any way with the retrieval process, so in fact, the token is valid for any arbitrary chain, wdyt?

If you think an auth-related test is still needed, I think we could include it in [this refactor](https://github.com/safe-global/safe-client-gateway/pull/1799) if you are OK with it.

Reply:
Indeed it does cover it - thanks for the explanation!